### PR TITLE
Angular 5: non-javascript changes

### PR DESCRIPTION
**Pull request by chouseknecht**
All the changes made to backend, admin or other supporting files to accommodate the move to Angular 5 and a single page application (SPA) framework.

### CODE REVIEW

File: galaxy/api/views/account.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import logging
+
+from rest_framework.status import HTTP_200_OK
```
Comment (cutwater) [inline galaxy/api/views/account.py]: It's usually recommended to import modules rather then module attributes.

File: galaxy/main/urls.py [hunk 2]
```diff
@@ -25,40 +25,28 @@
...
     urlpatterns += [
-        url(r'^static/(?P<path>.*)$',
-            never_cache(serve_staticfiles))
+        url(r'^static/(?P<path>.*)$', never_cache(serve_staticfiles))
     ]
 else:
     urlpatterns += [
-        url(r'^static/(?P<path>.*)$', serve_static,
-            kwargs={'document_root': settings.STATIC_ROOT})
+        url(r'^static/(?P<path>.*)$',
```
Comment (cutwater) [inline galaxy/main/urls.py]: Can you please explain this change?

File: galaxy/main/views.py [hunk 3]
```diff
@@ -129,16 +129,16 @@ def build_standard_context(request):
...
-    context = build_standard_context(request)
-    contentblocks = ContentBlock.objects.filter(name__in=[
-        'main-title',
-        'main-share',
-        'main-downloads',
-        'main-featured-blog'
-    ]).all()
-    context['contentblocks'] = {item.name: item for item in contentblocks}
-    return render_to_response('home.html', context)
+# def home(request):
```
Comment (cutwater) [inline galaxy/main/views.py]: Please delete unneeded code.

File: galaxy/main/urls.py (hunk 2 above)
Comment (chouseknecht) [reply galaxy/main/urls.py]: Looks like a formatting change.

**Comment by chouseknecht** [comment]
Tests aren't going to pass, because the Makefile is based on the existence of `galaxyui`, and all of the new javascript. However, none of those files are included in this PR.

File: galaxy/main/urls.py (hunk 2 above)
Comment (chouseknecht) [reply galaxy/main/urls.py]: Maybe what you're really asking about is the 2 lines appearing after your comment, which are the following:

```
    url(r'^(?P<path>.*[css|js|png|jpg|jpeg|ico|woff|woff2|svg|ttf])/?$',
        serve_static, kwargs={'document_root': settings.STATIC_ROOT}),
    url(r'^$',
        serve_static, kwargs={'document_root': settings.STATIC_ROOT, 'path': 'index.html'})
```

When `DEBUG=False`, any requests that don't match a regex pattern are likely part of the Angular app. When deployed, the Angular files (including all the enumerated file types) are deployed to `STATIC_ROOT`. However, the request will not include `/static`, so the above two patterns make sure the server try to find the requested file within the static directory.

**Pull request merged**